src/tools/lint-docs/src/lib.rs

```diff
@@ -441,10 +441,14 @@ impl<'a> LintExtractor<'a> {
         fs::write(&tempfile, source)
             .map_err(|e| format!("failed to write {}: {}", tempfile.display(), e))?;
         let mut cmd = Command::new(self.rustc_path);
-        if options.contains(&"edition2015") {
-            cmd.arg("--edition=2015");
-        } else {
+        if options.contains(&"edition2024") {
+            cmd.arg("--edition=2024");
+        } else if options.contains(&"edition2021") {
+            cmd.arg("--edition=2021");
+        } else if options.contains(&"edition2018") {
             cmd.arg("--edition=2018");
+        } else {
+            cmd.arg("--edition=2015");
```

Comment:
The edition was intentionally set to 2018 to keep it on the "latest" edition. It just never got updated for 2021. I think it would be good to keep the behavior where the most recent edition (2021, or whatever the compiler is built with) is the default.